tests/test_SatfuncConsistencyCheckManager.cpp

```diff
@@ -0,0 +1,1612 @@
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <utility>
+
+#include <fmt/format.h>
+
+#if HAVE_MPI
+#include <array>
+#include <iostream>
+#endif  // HAVE_MPI
+
+// ###########################################################################
+
+namespace {
+
+#if HAVE_MPI
+    struct MPIError
+    {
+        MPIError(std::string_view errstr, const int ec)
+            : errorstring { errstr }
+            , errorcode   { ec }
+        {}
+
+        std::string errorstring;
+        int errorcode;
+    };
+
+    void MPI_err_handler(MPI_Comm*, int* err_code, ...)
+    {
+        std::array<char, MPI_MAX_ERROR_STRING> err_string_vec{'\0'};
+        auto err_length = 0;
+
+        MPI_Error_string(*err_code, err_string_vec.data(), &err_length);
+
+        auto err_string = std::string_view {
+            err_string_vec.data(),
+            static_cast<std::string_view::size_type>(err_length)
+        };
+
+        std::cerr << "An MPI Error ocurred:\n  -> " << err_string << '\n';
+
+        throw MPIError { err_string, *err_code };
+    }
+
+    // Register a throwing error handler to allow for debugging with
+    //
+    //   catch throw
+    //
+    // in GDB.
+    void register_error_handler()
+    {
+        MPI_Errhandler handler{};
+
+        MPI_Comm_create_errhandler(MPI_err_handler, &handler);
+        MPI_Comm_set_errhandler(MPI_COMM_WORLD, handler);
+    }
+
+#else // !HAVE_MPI
+
+    void register_error_handler()
+    {}
+
+#endif // HAVE_MPI
+
+    bool init_unit_test_func()
+    {
+        return true;
+    }
+
+    // -----------------------------------------------------------------------
+
+    template <typename Scalar>
+    using CheckMgr = Opm::Satfunc::PhaseChecks::SatfuncConsistencyCheckManager<Scalar>;
+
+    template <typename Scalar>
+    using ViolationLevel = typename CheckMgr<Scalar>::ViolationLevel;
+
+    constexpr auto root = 0;
+    constexpr auto numSamplePoints = std::size_t{1};
+    const auto localToGlobal = [](const int) { return std::size_t{0}; };
+
+    std::pair<Dune::CpGrid, Opm::EclipseState>
+    setup(std::string_view epsSpec)
+    {
+        auto ret = std::pair<Dune::CpGrid, Opm::EclipseState> {
+            std::piecewise_construct,
+            std::forward_as_tuple(),
+            std::forward_as_tuple(Opm::Parser{}.parseString(fmt::format(R"(RUNSPEC
+DIMENS
+ 1 1 1 /
+
+OIL
+GAS
+WATER
+
+TABDIMS
+/
+
+ENDSCALE
+/
+
+-- =================================================================
+GRID
+
+DXV
+ 100 /
+
+DYV
+ 100 /
+
+DZV
+ 5 /
+
+DEPTHZ
+ 4*2000 /
+
+PERMX
+ 100 /
+
+PERMY
+ 100 /
+
+PERMZ
+ 10 /
+
+PORO
+ 0.3 /
+
+-- =================================================================
+PROPS
+
+SGOF
+ 0.00  0.0  1.0 0.0
+ 0.80   1*  0.0 0.0
+ 0.85  1.0  0.0 0.0
+ /
+
+SWOF
+ 0.15  0.0  1.0 0.0
+ 0.80  0.8  0.0 0.0
+ 1.0   1.0  0.0 0.0
+/
+{}
+
+-- =================================================================
+REGIONS
+
+SATNUM
+ 1 /
+
+END
+)", epsSpec)))
```

Comment:
This neat trick was a bit hard to notice, would be better to put this string generation into a separate function.

Reply:
> This neat trick was a bit hard to notice, would be better to put this string generation into a separate function.

I've added a helper function that returns a `Deck` object instead.  I didn't see any benefit to having a function of the form
```C++
std::string makeEpsString(std::string_view epsSpec)
{
    return fmt::format(R"(RUNSPEC
...)", epsSpec);
}
```
as the return value would just be passed immediately into `Parser{}.parseString()`.